cmd/crates/soroban-spec-typescript/src/lib.rs

```diff
@@ -91,64 +116,100 @@ pub fn generate(spec: &[ScSpecEntry]) -> String {
-         *   - `'simulated'` will only simulate/preflight the transaction, even if it's a change/set method that requires auth/signing. Returns full preflight info.
-         *   - `'full'` return the full RPC response, meaning either 1. the preflight info, if it's a view/read method that doesn't require auth/signing, or 2. the `sendTransaction` response, if there's a problem with sending the transaction or if you set `secondsToWait` to 0, or 3. the `getTransaction` response, if it's a change method with no `sendTransaction` errors and a positive `secondsToWait`.
-         */
-        responseType?: R
-        /**
-         * If the simulation shows that this invocation requires auth/signing, `invoke` will wait `secondsToWait` seconds for the transaction to complete before giving up and returning the incomplete {{@link SorobanClient.SorobanRpc.GetTransactionResponse}} results (or attempting to parse their probably-missing XDR with `parseResultXdr`, depending on `responseType`). Set this to `0` to skip waiting altogether, which will return you {{@link SorobanClient.SorobanRpc.SendTransactionResponse}} more quickly, before the transaction has time to be included in the ledger. Default: 10.
-         */
-        secondsToWait?: number
-    }}"#
-    )
-}
+        fee?: number,
+    }"#;
 
 fn jsify_name(name: &String) -> String {
     name.to_lower_camel_case()
 }
 
+pub fn entry_to_parser(entry: &Entry) -> Option<(String, String)> {
+    if let Entry::Function { name, outputs, ..  } = entry {
+        let mut is_result = false;
+        let mut return_type: String;
+        if outputs.is_empty() {
+            return_type = "void".to_owned();
+        } else if outputs.len() == 1 {
+            return_type = type_to_ts(&outputs[0]);
+            is_result = return_type.starts_with("Result<");
+        } else {
+            return_type = format!("readonly [{}]", outputs.iter().map(type_to_ts).join(", "));
+        };
+
+        if is_result {
+            return_type = return_type
+                .strip_prefix("Result<")
+                .unwrap()
+                .strip_suffix('>')
+                .unwrap()
+                .to_owned();
+            return_type = format!("Ok<{return_type}> | Err<Error_>");
+        }
+
+        let mut output = outputs
+            .get(0)
+            .map(|_| format!("this.spec.funcResToNative(\"{name}\", xdr)"))
+            .unwrap_or_default();
+        if is_result {
+            output = format!("new Ok({output})");
+        }
+        let parse_result_xdr = if return_type == "void" {
+            r#"() => {}"#.to_owned()
+        } else if is_result {
+            format!(r#"(result: XDR_BASE64 | Err): {return_type} => {{
+            if (result instanceof Err) return result
+            return new Ok(this.spec.funcResToNative("u32_fail_on_even", result))
```

Comment:
what purpose is `u32_fail_on_even` here for? :confused:

Reply:
copypasta 🤦🏻